.github/workflows/build.yml

```diff
@@ -25,29 +25,39 @@ on:
       - '.github/workflows/build.yml'
 
 jobs:
-
   build:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        include:
+          - os: ubuntu-latest
+            cc: gcc
+          - os: macos-latest
+            cc: clang
```

Comment:
Apart from adding macos to the CI, I like the clang here for other reasons also. I was already thinking to add clang builds to the CI, so there it is. Done. Thanks! :-)

Reply:
It isn't fully matrix'ed though. I'ts only mapping gcc to ubuntu and clang to macos. We can add cc to the matrix to get all four combinations though!